pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1011,11 +1041,17 @@ func (ctrl *PersistentVolumeController) reclaimVolume(volume *v1.PersistentVolum
 	case v1.PersistentVolumeReclaimDelete:
 		klog.V(4).Infof("reclaimVolume[%s]: policy is Delete", volume.Name)
 		opName := fmt.Sprintf("delete-%s[%s]", volume.Name, string(volume.UID))
-		startTime := time.Now()
+		// create a start timestamp entry in cache for deletion operation if no one exists with
+		// key = volume.Name, pluginName = provisionerName, operation = "delete"
+		ctrl.operationTimestamps.AddIfNotExist(volume.Name, ctrl.getProvisionerNameFromVolume(volume), "delete")
```

Comment:
We aren't handling translated names correctly for delete operation - are we? it seems that, we should apply same logic we apply for provisioning operation. cc @davidz627

Reply:
the provisioner name for deletion is currently obtained from storage class's "provisioner" field. For provisioning, from my observation via e2e test, the same field is used for as the annotation "storage-provisioner" value for a PVC, as well as the annotation "provisioned-by" value of a PV.

Reply:
Looks like `ctrl.getProvisionerName` does a translation if the volume is migrated, but `getProvisionerNameFromVolume` does not, they both should. Alternatively, could we just use: the `"pv.kubernetes.io/provisioned-by"` annotation which will be set correctly by migration path, instead of `GET`-ing the storage class inside these helper functions.

Reply:
@davidz627 for PV, is annotation "volume.beta.kubernetes.io/storage-provisioner" reliable enough as the annotation "pv.kubernetes.io/provisioned-by" for PV?